### DISCUSSION TITLE
Suggestion: create a codespaces quickstart for fastapi contributors

### DISCUSSION THREAD

**Discussion by Ryandaydev** [fastapi/fastapi#10714] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-python-project-for-codespaces
```

### Description

This is something to consider to help contributors get up and running faster. I suggest we create (and maintain) a FastAPI contributors quickstart that contains the basic dependencies needed to contribute to the fastAPI code. I have seen people adding codespaces for using the software (including GitHub as one of their first codespaces tutorials) but haven't seen a dedicated one for contributing and using in dev mode.

This could replace/enhance the setup instructions here: https://fastapi.tiangolo.com/contributing/

A couple of benefits of using this approach (which I see more open source projects adopting):

1. It accelerates new contributors setting up their environments. (Especially helpful for the language translations that I know are important to @tiangolo.

2. It's a super easy way to keep the dev copy of the code separate from the working copy (this is something that has caused me a headache once or twice).

I'll raise my hand to help out getting this set up if it's something we want to consider.

### Operating System

Linux

### FastAPI Version

0.103.2

### Pydantic Version

2.3.0

### Python Version

3.10

### Additional Context

Here's an example of the Apache Airflow project using this method:
https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START_CODESPACES.rst

**Comment by Kludex** [comment, marked as answer]
https://github.com/tiangolo/fastapi/pull/9262#issuecomment-1738109808

**Reply by Ryandaydev** [reply]
Thanks @Kludex I missed that one in the search. Added a comment to make it easier to find next time.
